src/core/event/entity_context.cpp

```diff
@@ -167,3 +175,21 @@ void entity_context::close_socket_job(const job_desc &job)
         close_socket_helper(tcp_si);
     }
 }
+
+/*static*/
+void entity_context::entity_context_comp_cb(xlio_socket_t sock, uintptr_t userdata_sq,
+                                            uintptr_t userdata_op)
+{
+    mem_buf_desc_t *buf_list = reinterpret_cast<mem_buf_desc_t *>(userdata_op);
+
+    NOT_IN_USE(sock);
+    NOT_IN_USE(userdata_sq);
+
+    if (likely(buf_list->p_desc_owner)) {
+        buf_list->p_desc_owner->mem_buf_tx_release(buf_list, true);
+    } else {
+        vlog_printf(VLOG_WARNING,
```

Comment:
Can we really have this situation?

Reply:
i guess yes. Without the check the code clears that there should always be an owner otherwise its a bug